### PR TITLE
Mask sensitive key values from module result

### PR DESCRIPTION
**Pull request by GomathiselviS**
Signed-off-by: GomathiselviS <gomathiselvi@gmail.com>

##### SUMMARY

Fixes #https://github.com/ansible-collections/arista.eos/issues/186

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by NilashishC — CHANGES_REQUESTED** [review]
The validate-modules:no-log-needed in ignore-2.11 list can be now be removed.

**Comment by NilashishC** [comment]
@GomathiselviS this would also need a bump in ansible.netcommon version in galaxy.yml to >=2.0.1. Please refer to https://github.com/ansible-collections/cisco.nxos/pull/278/files#diff-1ade2795a7b410581d689fb0faa358b769c2d189b2d549e8f530b706f09b2fb7R5.

**Comment by GomathiselviS** [comment]
recheck

File: plugins/module_utils/network/eos/argspec/ospf_interfaces/ospf_interfaces.py [hunk 1]
```diff
@@ -109,9 +109,9 @@ def __init__(self, **kwargs):
                                         "null",
                                     ],
                                 },
-                                "keytype": {"type": "str"},
+                                "keytype": {"type": "str", "no_log": True},
```
Comment (NilashishC) [inline plugins/module_utils/network/eos/argspec/ospf_interfaces/ospf_interfaces.py:112]: Does keytype sensitive data? If not then we can set no_log: False explicitly to bypass validate-modules.

File: plugins/module_utils/network/eos/argspec/ospf_interfaces/ospf_interfaces.py [hunk 2]
```diff
@@ -131,10 +131,10 @@ def __init__(self, **kwargs):
                         "authentication_v3": {
                             "type": "dict",
                             "options": {
-                                "key": {"type": "str"},
+                                "key": {"type": "str", "no_log": True},
                                 "spi": {"type": "int"},
-                                "keytype": {"type": "str"},
-                                "passphrase": {"type": "str"},
+                                "keytype": {"type": "str", "no_log": True},
```
Comment (NilashishC) [inline plugins/module_utils/network/eos/argspec/ospf_interfaces/ospf_interfaces.py:136]: Same as above.

File: plugins/module_utils/network/eos/argspec/ospf_interfaces/ospf_interfaces.py [hunk 3]
```diff
@@ -143,10 +143,11 @@ def __init__(self, **kwargs):
                         },
                         "retransmit_interval": {"type": "int"},
                         "message_digest_key": {
+                            "no_log": True,
```
Comment (NilashishC) [inline plugins/module_utils/network/eos/argspec/ospf_interfaces/ospf_interfaces.py:146]: Same discussion as in VyOS PR.

**Review by NilashishC — APPROVED** [review]

**Pull request merged**